peps/pep-9999.rst

```diff
@@ -0,0 +1,616 @@
+be collected into their respective groups, and these groups should be evaluated
+in the same order as environment variables and configuration files.
+
+For example, the CLI example given above will evaluate with pseudocode:
+
+::
+
+   --extra-index-url <highest> --find-links <next_priority> --extra-index-url <lowest>
+   # ------------
+   find_links = []
+   extra_index_urls = []
+   for argname, value in cli_args:
+     match argname:
+       case "find-links":
+         find_links.append(value)
+       case "extra-index-url":
+         extra_index_urls.append(value)
+   search_urls = find_links + extra_index_urls
+
+
+The ultimate result of this evalution will be:
+
+1. next_priority
+2. highest
+3. lowest
+
+The value of ``index-url`` (or its default value of PyPI) is always the
+lowest-priority entry in the search order.
+
+Requirements.txt file inclusions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As a further complication, `requirements.txt files can include index url
+options <https://pip.pypa.io/en/stable/reference/requirements-file-format/#global-options>`__,
+including ``--extra-index-url`` and ``--find-links``. Requirements.txt
+can be included via an environment variable (PIP_REQUIREMENT) and `by
+command line
+arguments <https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-r>`__.
+For the purposes of evaluating priority, these requirements files should
+be evaluated where they are specified (i.e. their position on the CLI
+args), and place their index url options at the priority of the
+originating file inclusion (either the env var or CLI args). For
+example, if a requirements.txt file has:
+
+::
+
+   --extra-index-url <req_file_index_url_1>
+   --find-links <some_local_path>
+   --extra-index-url <req_file_index_url_2>
+   my-package
+
+The CLI command of:
+
+::
+
+   pip install --extra-index-url <something> -r requirements.txt --extra-index-url <another>
+
+Would expand to an index priority list of:
+
+1. <some_local_path>
+2. <something>
+3. <req_file_index_url_1>
+4. <req_file_index_url_2>
+5. <another>
+6. <fallback to value of index-url>
+
+Because the requirements.txt file is parsed from top to bottom, it
+evaluates ``--extra-index-url`` and ``--find-links`` arguments in the
+order that they appear in the requirements.txt file, as opposed to
+reordering them such that ``--find-links`` arguments always come before
+any ``--extra-index-url`` arguments. Requirements files may also include
+other requirements files. In the general case, depth-first
+traversal of requirements files defines the collection and ordering
+of index-urls and find-links options.
+
+Proposed deny list format
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As described above, `it may be desirable to intentionally omit one or
+more packages from consideration on a given
+index <https://github.com/astral-sh/uv/issues/4753>`__. For example, if
+a repository includes builds of a package that introduces issues, but
```

Comment:
The PEP uses "repository" and "index" somewhat interchangeably and inconsistently, but I think it's better to converge on the term "index" for what PyPI-like things are.

Reply:
I replaced all instances of repo with index. I agree that loosey-goosey flipping is not helpful here.